exchange/utils.go

```diff
@@ -760,7 +760,8 @@ func applyFPD(fpd *firstpartydata.ResolvedFirstPartyData, bidReq *openrtb2.BidRe
 
 func buildBidResponseRequest(req *openrtb2.BidRequest,
 	bidderImpResponses stored_responses.BidderImpsWithBidResponses,
-	aliases map[string]string) map[openrtb_ext.BidderName]BidderRequest {
+	aliases map[string]string,
```

Comment:
I realize that the code discussed in this comment wasn't added in your PR, but I believe we could address it in order to simplify both this function and `cleanOpenRTBRequests()`. The return value of `stored_responses.InitStoredBidResponses(req.BidRequest, auctionReq.StoredBidResponses)` seems to not be used inside `cleanOpenRTBRequests()` until line 76 in the `buildBidResponseRequest()` call and not used again.  In an effort to make `cleanOpenRTBRequests()` less verbose, can we move the call inside `buildBidResponseRequest()` instead?
```
 40   func cleanOpenRTBRequests(ctx context.Context,
 41       auctionReq AuctionRequest,
 42       requestExt *openrtb_ext.ExtRequest,
 43       bidderToSyncerKey map[string]string,
 44       metricsEngine metrics.MetricsEngine,
 45       gdprDefaultValue gdpr.Signal,
 46       privacyConfig config.Privacy,
 47       gdprPermsBuilder gdpr.PermissionsBuilder,
 48       tcf2ConfigBuilder gdpr.TCF2ConfigBuilder,
 49       hostSChainNode *openrtb2.SupplyChainNode,
 50   ) (allowedBidderRequests []BidderRequest, privacyLabels metrics.PrivacyLabels, errs []error) {
 51  
 52       req := auctionReq.BidRequestWrapper
 53       aliases, errs := parseAliases(req.BidRequest)
 54       if len(errs) > 0 {
 55           return
 56       }
 57  
 58       allowedBidderRequests = make([]BidderRequest, 0, 0)
 59  
 60 -     bidderImpWithBidResp := stored_responses.InitStoredBidResponses(req.BidRequest, auctionReq.StoredBidResponses)
 61  
 62       impsByBidder, err := splitImps(req.BidRequest.Imp)
 63       if err != nil {
 64           errs = []error{err}
 65           return
 66       }
 67  
 68       aliasesGVLIDs, errs := parseAliasesGVLIDs(req.BidRequest)
 69       if len(errs) > 0 {
 70           return
 71       }
 72  
 73       var allBidderRequests []BidderRequest
 74       allBidderRequests, errs = getAuctionBidderRequests(auctionReq, requestExt, bidderToSyncerKey, impsByBidder, aliases, hostSChainNode)
 75  
 76 -     bidderNameToBidderReq := buildBidResponseRequest(req.BidRequest, bidderImpWithBidResp, aliases, auctionReq.BidderImpReplaceImp)
    +     bidderNameToBidderReq := buildBidResponseRequest(req.BidRequest, aliases, auctionReq.BidderImpReplaceImp)
 77       //this function should be executed after getAuctionBidderRequests
 78       allBidderRequests = mergeBidderRequests(allBidderRequests, bidderNameToBidderReq)
 78     allBidderRequests = mergeBidderRequests(allBidderRequests, bidderNameToBidderReq)
 79
 80 *--677 lines: gdprSignal, err := extractGDPR(req.BidRequest)---------------------------------------------------------------------------------------------------------------------------------------------------
757           bidReq.User = fpd.User
758       }
759   }
760
761   func buildBidResponseRequest(req *openrtb2.BidRequest,
762 -     bidderImpResponses stored_responses.BidderImpsWithBidResponses,
763       aliases map[string]string,
764       bidderImpReplaceImpId stored_responses.BidderImpReplaceImpId) map[openrtb_ext.BidderName]BidderRequest {
    +     
    +     bidderImpWithBidResp := stored_responses.InitStoredBidResponses(req.BidRequest, auctionReq.StoredBidResponses)
    +     
765       bidderToBidderResponse := make(map[openrtb_ext.BidderName]BidderRequest)
766 -     for bidderName, impResps := range bidderImpResponses {
    +     for bidderName, impResps := range bidderImpWithBidResp {
767           resolvedBidder := resolveBidder(string(bidderName), aliases)
768           bidderToBidderResponse[bidderName] = BidderRequest{
769               BidRequest:            req,
770               BidderCoreName:        resolvedBidder,
771               BidderName:            bidderName,
772               BidderStoredResponses: impResps,
773               ImpReplaceImpId:       bidderImpReplaceImpId[string(resolvedBidder)],
774               BidderLabels:          metrics.AdapterLabels{Adapter: resolvedBidder},
775           }
776       }
777       return bidderToBidderResponse
778   }
exchange/utils.go
```
Do you think addressing this in this PR is worth it?

Reply:
Oh I see what you mean. 
This is done on purpose. Imagine the situation when we have both real impressions and impressions with stored responses in one request. They can belong to different bidders or the same bidder. 
In InitStoredBidResponses function we extract imps with stored responses and remove them from request. Then we execute existing logic to split reminding  imps by bidder and then we combine them all together to bidder request per bidder. 
As a result we come up with `BidderRequest` where we clearly see what imps should go to MakeRequests and in `BidderStoredResponses` we store imps with stored responses.

Hope it makes sense.